orangecontrib/geo/widgets/owchoropleth.py

```diff
@@ -710,6 +711,18 @@ def set_data(self, data):
         self.apply_selection()
         self.unconditional_commit()
 
+    def check_data(self):
+        self.Warning.no_region.clear()
+        self.Warning.no_lat_lon_vars.clear()
+        self._attr_lat, self._attr_lon = None, None
+        if self.data is not None:
+            self._attr_lat, self._attr_lon = \
```

Comment:
How would you describe the attributes `_attr_lat` and `_attr_lon`? What's their relation with `attr_lat` and `attr_lon`? If you can't name them, I don't like them; in particular I don't like them being attributes. As I understand, they are "transient attributes", into which `check_data` stores the variables before `init_attr_value` copies them to "real" attributes `attr_lat` and `attr_lon`. Can you reorganize the code so that it doesn't (ab)use attributes for passing data from `check_data` to `init_attr_values`, in particular because both methods are called from a single place and just one after the other? This may not be the best solution, but: you can just merge `check_data` into `init_attr_values`... Or are these methods called from super, somewhere?

Reply:
I will merge them. I just wanted to look nicer, apparently the opposite happened. Those transient attributes are there so I don't call the find_lat_lon twice in different functions.